src/Analyzer/Passes/ConvertOrLikeChainPass.cpp

```diff
@@ -1,54 +1,52 @@
+#include <Analyzer/Passes/ConvertOrLikeChainPass.h>
+
 #include <memory>
 #include <unordered_map>
 #include <vector>
-#include <Analyzer/Passes/ConvertOrLikeChainPass.h>
-#include <Analyzer/ConstantNode.h>
-#include <Analyzer/UnionNode.h>
-#include <Analyzer/FunctionNode.h>
-#include <Analyzer/HashUtils.h>
-#include <Analyzer/InDepthQueryTreeVisitor.h>
+
 #include <Core/Field.h>
+
 #include <DataTypes/DataTypesNumber.h>
+
 #include <Functions/FunctionFactory.h>
 #include <Functions/likePatternToRegexp.h>
+
 #include <Interpreters/Context.h>
 
+#include <Analyzer/ConstantNode.h>
+#include <Analyzer/UnionNode.h>
+#include <Analyzer/FunctionNode.h>
+#include <Analyzer/HashUtils.h>
+#include <Analyzer/InDepthQueryTreeVisitor.h>
+
 namespace DB
 {
 
 namespace
 {
 
-class ConvertOrLikeChainVisitor : public InDepthQueryTreeVisitor<ConvertOrLikeChainVisitor>
+class ConvertOrLikeChainVisitor : public InDepthQueryTreeVisitorWithContext<ConvertOrLikeChainVisitor>
 {
-    using FunctionNodes = std::vector<std::shared_ptr<FunctionNode>>;
-
-    const FunctionOverloadResolverPtr match_function_ref;
-    const FunctionOverloadResolverPtr or_function_resolver;
 public:
-
-    explicit ConvertOrLikeChainVisitor(ContextPtr context)
-        : InDepthQueryTreeVisitor<ConvertOrLikeChainVisitor>()
-        , match_function_ref(FunctionFactory::instance().get("multiMatchAny", context))
-        , or_function_resolver(FunctionFactory::instance().get("or", context))
+    using Base = InDepthQueryTreeVisitorWithContext<ConvertOrLikeChainVisitor>;
+    using Base::Base;
+
+    explicit ConvertOrLikeChainVisitor(FunctionOverloadResolverPtr or_function_resolver_,
+        FunctionOverloadResolverPtr match_function_resolver_,
+        ContextPtr context)
+        : Base(std::move(context))
+        , or_function_resolver(std::move(or_function_resolver_))
+        , match_function_resolver(std::move(match_function_resolver_))
     {}
 
-    static bool needChildVisit(VisitQueryTreeNodeType & parent, VisitQueryTreeNodeType &)
+    bool needChildVisit(VisitQueryTreeNodeType &, VisitQueryTreeNodeType &)
```

Comment:
This predicate is not evaluated for the root node, right?

Reply:
As we discussed current visitor interface need to be improved later.